py_gtfs_rt_ingestion/dev_test_setup.py

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env python
+
+import argparse
+import logging
+import sys
+from typing import NamedTuple
+import os
+import random
+from multiprocessing import Pool
+from collections.abc import Iterable
+
+import boto3
+
+from py_gtfs_rt_ingestion import file_list_from_s3
+
+GTFS_BUCKET = "mbta-gtfs-s3"
+DEV_INGEST_BUCKET = "mbta-ctd-dataplatform-dev-incoming"
+DEV_EXPORT_BUCKET = "mbta-ctd-dataplatform-dev-springboard"
+DEV_ARCHIVE_BUCKET = "mbta-ctd-dataplatform-dev-archive"
+DEV_ERROR_BUCKET = "mbta-ctd-dataplatform-dev-error"
+LAMP_PREFIX = "lamp/"
+
+logging.basicConfig(level=logging.WARNING)
+
+
+class SetupArgs(NamedTuple):
+    """
+    NamedTuple to hold Setup arguments
+    """
+
+    src_prefix: str
+    objs_to_copy: int
+    force_delete: bool
+
+
+def parse_args(args: list[str]) -> SetupArgs:
+    """
+    parse input args from the command line and generate an event dict in the
+    format the lambda handler is expecting
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-s",
+        "--source-prefix",
+        dest="src_prefix",
+        type=str,
+        default="",
+        help="prefix of objects to move into dev environment",
+    )
+
+    parser.add_argument(
+        "-n",
+        "--num",
+        dest="objs_to_copy",
+        type=int,
+        default=1_000,
+        help="number of objects to move into dev environment",
+    )
+
+    parser.add_argument(
+        "-f",
+        dest="force_delete",
+        default=False,
+        action="store_true",
+        help="delete objects from dev buckets without asking",
+    )
+
+    return SetupArgs(**vars(parser.parse_args(args)))
+
+
+def del_objs(del_list: list[str], bucket: str) -> int:
+    """
+    Delete del_list of objects from bucket
+    """
+    s3_client = boto3.client("s3")
+    delete_objs = {
+        "Objects": [
+            {"Key": uri.replace(f"s3://{bucket}/", "")} for uri in del_list
+        ]
+    }
+    try:
+        s3_client.delete_objects(
+            Bucket=bucket,
+            Delete=delete_objs,
+        )
+    except Exception as e:
+        logging.exception(e)
+        return 0
+    return len(del_list)
```

Comment:
Function to delete batches of objects. The boto3 library limits the size of these to 1,000 max.